File: 프로그래머스/unrated/159994. 카드 뭉치/카드 뭉치.js
function solution(cards1, cards2, goal) {
    let result = 'Yes';
    
    cards1.reverse();
    cards2.reverse(); // pop을 사용하기 위해.. 
    
    for(let i = 0; i < goal.length; i++) {
        if(cards1[cards1.length - 1] === goal[i]) cards1.pop();
        else if(cards2[cards2.length - 1] === goal[i]) cards2.pop();
        else {
            result = 'No';
            break;
        }
    }
    
    return result;
}

File: 프로그래머스/1/159994. 카드 뭉치/카드 뭉치.py
'''
카드 스택
O(n+n)
두 개의 top을 보면 됨. 만들 수 있는 지만 찾으면 되기 때문에 스택에 남은게 있어도 됨. 
'''
from collections import deque
def solution(cards1, cards2, goal):
    left = deque(cards1)
    right = deque(cards2)
    goal = deque(goal)
    
    res = 0
    
    while (True):
        if len(goal) == 0:
            return 'Yes'
        
        ltop = None
        if len(left) > 0:
            ltop = left[0]
            
        rtop = None
        if len(right) > 0:
            rtop = right[0]
            
        if ltop == goal[0]:
            left.popleft()
            goal.popleft()
            continue
        
        elif rtop == goal[0]:
            right.popleft()
            goal.popleft()
            continue
        
        return 'No'
            
        
        
            
        
